docs/chapter4-payload-handoff-format.rst

```diff
@@ -29,9 +29,834 @@ architecture, either 32 or 64 bits.
+[DTspec]_.
+
+This node is mandatory if a serial device is available, whether or not if it is
+enabled by Platform Init.
+
+.. tabularcolumns:: | p{3cm} p{0.75cm} p{2cm} p{9.5cm} |
+.. table:: Node: serial@<addr>
+
+   ================= =========== ================== ======================================================================================================================================================================
+   Property Name     Usage       Value Type         Definition
+   ================= =========== ================== ======================================================================================================================================================================
+   compatible        R           string list        Compatible string for hardware.
+                                                    Currently, these compatible strings are supported:
+                                                    ns16550a, ns16550, ns8250, ns16450
+   clock-frequency   R           u32                Frequency (in Hz) of the baud rate generator’s input clock.
+   current-speed     R           u32                Current serial device speed in bits per second.
+   reg               R           prop-encoded-array Physical address of the registers device within the address space of the parent. The form of reg is
+                                                    <[address] [size]>
+   reg-shift         O           u32                log2 of distance between the discrete device registers. If unspecified, the default value is 0, meaning 1 byte apart.
+   reg-offset        O           u32                Offset of the registers from the base address. The default value is 0, meaning no offset.
+   reg-io-width      O           u32                Register width in bytes. Valid values are 1, 2 and 4. The default value is 1, meaning byte width.
+   virtual-reg       SD          u32 or u64         Specifies an effective address that maps to the first physical address specified in the reg property. This property is
+                                                    required if this device node is the system’s console.
+   ================= =========== ================== ======================================================================================================================================================================
+
+
+Example below shows a legacy I/O serial device and an MMIO serial device:
+
+.. code-block:: dts
+
+   
+   // Legacy I/O serial device
+   isa {
+      compatible = "isa";
+      #address-cells = <2>;
+      #size-cells = <1>;
+
+      serial@3f8 {
+         compatible = "ns16550";
+         reg-io-width = <1>;
+         reg = <0x3f8 8>;
+         clock-frequency = <0x1c2000>;
+         current-speed = <115200>;
+      };
+   };
+
+   // MMIO serial device
+   serial@fe037000 {
+      compatible = "ns16550a";
+      reg-io-width = <4>;
+      reg = /bits 64/ <0xfe037000 0x80>;
```

Comment:
I recalled we would like all pci devices reg/ranges to self-describe address is MMIO or IO, so here reg should be like below:
<0x82000000 0 0xfe037000 0 0x80> //non-prefetchable, non-relocable, non-aliased 32bit MMIO

Reply:
Since this exists under soc/ root node, not under PCI node, so the reg has to follow the root node address and size cells, hence this example is valid. Should i add another example for serial under PCI?

Reply:
Sounds good idea.

Reply:
Done.